crates/red_knot_python_semantic/src/types/narrow.rs

```diff
@@ -327,6 +334,12 @@ impl<'db> NarrowingConstraintsBuilder<'db> {
                                 constraints.insert(symbol, ty);
                             }
                         }
+                        ast::CmpOp::Eq if lhs_ty.is_literal_string() => {
+                            let ty = IntersectionBuilder::new(self.db)
+                                .add_positive(rhs_ty)
+                                .build();
+                            constraints.insert(symbol, ty);
```

Comment:
Building an intersection with just one type in it is a no-op, just use the single type directly.

```suggestion
                            constraints.insert(symbol, rhs_ty);
```